### PR TITLE
first version of solution

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

File: src/main/java/com/dev/movietheatre/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -43,9 +43,9 @@ public User add(User user) {
     @Override
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
-            Query<User> findByEmailQuery = session.createQuery("from User u "
-                    + " where email = :user_email", User.class);
-            findByEmailQuery.setParameter("user_email", email);
+            Query<User> findByEmailQuery = session.createQuery("SELECT u FROM User u "
+                    + " LEFT JOIN FETCH u.roles WHERE u.email = :email", User.class);
```
Comment (AndreiYazadzhi) [inline src/main/java/com/dev/movietheatre/dao/impl/UserDaoImpl.java]: Use inner join

File: src/main/java/com/dev/movietheatre/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -55,7 +55,10 @@ public User add(User user) {
     @Override
     public Optional<User> getById(Long id) {
         try (Session session = sessionFactory.openSession()) {
-            return Optional.ofNullable(session.get(User.class, id));
+            return session.createQuery("SELECT u FROM User u"
+                    + " LEFT JOIN FETCH u.roles WHERE u.id = :id", User.class)
```
Comment (AndreiYazadzhi) [inline src/main/java/com/dev/movietheatre/dao/impl/UserDaoImpl.java]: Same

File: src/main/java/com/dev/movietheatre/model/Role.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+@Entity
+@Table(name = "roles")
+public class Role {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Enumerated
+    @Column(nullable = false, unique = true)
+    private Roles roleName;
```
Comment (AndreiYazadzhi) [inline src/main/java/com/dev/movietheatre/model/Role.java]: add column name

File: src/main/java/com/dev/movietheatre/security/impl/CustomUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        User user = userService.findByEmail(userName)
+                .orElseThrow(() -> new UsernameNotFoundException("User was not found!"));
+
+        UserBuilder builder = org.springframework.security.core.userdetails.User.builder();
```
Comment (AndreiYazadzhi) [inline src/main/java/com/dev/movietheatre/security/impl/CustomUserDetailsService.java:24]: Move it to imports

**Review by AndreiYazadzhi — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/movietheatre/model/Role.java (hunk 3 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/model/Role.java]: done

**Review by AndreiYazadzhi — APPROVED** [review]

File: src/main/java/com/dev/movietheatre/model/Roles.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.dev.movietheatre.model;
+
+public enum Roles {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/movietheatre/model/Roles.java]:
```suggestion
public enum RoleName {
```

File: src/main/java/com/dev/movietheatre/security/impl/CustomUserDetailsService.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        User user = userService.findByEmail(userName)
+                .orElseThrow(() -> new UsernameNotFoundException("User was not found!"));
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/movietheatre/security/impl/CustomUserDetailsService.java]: add method input data to message to make more informative

File: src/main/java/com/dev/movietheatre/model/User.java [hunk 7]
```diff
@@ -16,6 +18,8 @@
     @Column(name = "email", unique = true, length = 20, nullable = false)
     private String email;
     private String password;
+    @ManyToMany
+    private List<Role> roles;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/movietheatre/model/User.java]: better to use Set here, to avoid duplication

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/movietheatre/model/User.java (hunk 7 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/model/User.java]: done

File: src/main/java/com/dev/movietheatre/security/impl/CustomUserDetailsService.java (hunk 6 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/security/impl/CustomUserDetailsService.java]: done

File: src/main/java/com/dev/movietheatre/model/Roles.java (hunk 5 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/model/Roles.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**